akd/src/directory.rs

```diff
@@ -464,6 +467,19 @@ impl<S: Database + 'static, V: VRFKeyStorage> Directory<S, V> {
             return Err(AkdError::Storage(StorageError::NotFound(msg)));
         }
 
+        #[cfg(feature = "preload_history")]
+        {
+            let mut lookup_infos = vec![];
+            for ud in user_data.iter() {
+                if let Ok(lo) = self.build_lookup_info(ud).await {
```

Comment:
@kevinlewi this *should* cover all of the nodes we'd need to walk in a history proof right? This already has the `ValueState`s so we should just be able to translate that to a `LookupInfo` which then lets us do a preload of the nodes we'd need to walk iiuc.